devito/mpi/distributed.py

```diff
@@ -421,6 +440,197 @@ def _rebuild(self, shape=None, dimensions=None, comm=None):
+                                   else np.zeros(0, dtype=np.int64)
+                                   if sdim_interval[dim].is_empty
+                                   else d[np.logical_and(d >= sdim_interval[dim].start,
+                                                         d <= sdim_interval[dim].end)]
+                                   if sdim_interval[dim].is_Interval
+                                   else np.array(sdim_interval[dim].args, dtype=np.int64)
+                                   for d in dec])
+
+        self._decomposition = [Decomposition(d, c)
+                               for d, c in zip(decompositions, self.mycoords)]
+
+        self._intervals = tuple((dist_interval[dim] if sdim_interval[dim] is None
+                                 else dist_interval[dim].intersect(sdim_interval[dim]))
+                                for dim in self.parent.dimensions)
+
+        # Keep track of the edges of the rank crossed by the SubDomain
+        # Format is {dimension: {side: crosses}} or {(sides,): crosses}
+        # (matches neighborhood)
+        crosses = {}
+        for dim in self.parent.dimensions:
+            sdim = self.dimension_map[dim]
+            if sdim.is_Sub:
+                in_rank = dist_interval[dim].issuperset(sdim_interval[dim])
+                off_rank = dist_interval[dim].isdisjoint(sdim_interval[dim])
+                if in_rank or off_rank:
+                    crosses[sdim] = {LEFT: False, RIGHT: False}
+                elif sdim.local:
+                    raise ValueError("SubDimension %s is local and cannot be"
+                                     " decomposed across MPI ranks" % dim)
+                else:
+                    crosses[sdim] = {LEFT: sdim_interval[dim].left
+                                     < dist_interval[dim].left,
+                                     RIGHT: sdim_interval[dim].right
+                                     > dist_interval[dim].right}
+            else:
+                crosses[sdim] = {LEFT: True, RIGHT: True}
+
+        for i in product([LEFT, CENTER, RIGHT], repeat=len(self.dimensions)):
+            crosses[i] = all(crosses[d][s] for d, s in zip(self.dimensions, i)
+                             if s in crosses[d])  # Skip over CENTER
+
+        self._crosses = frozendict(crosses)
+
+    @property
+    def parent(self):
+        """The parent distributor of this SubDomainDistributor"""
+        return self._parent
+
+    @property
+    def dimension_map(self):
+        return self._dimension_map
+
+    @property
+    def intervals(self):
+        """The interval spanned by the SubDomain in each dimension on this rank"""
+        return self._intervals
+
+    @property
+    def crosses(self):
+        """
+        A mapper ``M`` indicating the sides of this MPI rank crossed by the SubDomain.
+        Let
+
+            * ``d`` be a Dimension -- ``d0, d1, ..., dn`` are the dimensions of the parent
```

Comment:
double quotes "``" are a thing of the past; now it's all single quotes "`". This is being rolled out throughout the code base, incrementally